packages/malloy/src/lang/parse-malloy.ts

```diff
@@ -783,10 +785,80 @@ export abstract class MalloyTranslation {
     const attempt = this.translateStep.step(this, extendingModel);
     if (attempt.final) {
       this.finalAnswer = attempt;
+      this.buildImports();
     }
     return attempt;
   }
 
+  importAt(position: DocumentPosition): ImportLocation | undefined {
+    // Here we assume that imports DO NOT overlap. And then we do a binary
+    // search to find the one we're looking for.
+    const index = this.findIndexBefore(position);
+    if (index === this.imports.length) {
+      return undefined;
+    }
+    const imp = this.imports[index];
+    if (
+      imp.location.range.start.line <= position.line &&
+      imp.location.range.end.line >= position.line &&
+      (position.line !== imp.location.range.start.line ||
+        position.character >= imp.location.range.start.character) &&
+      (position.line !== imp.location.range.end.line ||
+        position.character <= imp.location.range.end.character)
+    ) {
+      return imp;
+    }
+    return undefined;
+  }
+
+  private buildImports(): void {
+    for (const [key, value] of Object.entries(this.root.importZone.location)) {
+      this.imports.push({importURL: key, location: value});
+    }
+    // Sort list of imports by end position so that we can do a binary
+    // search to find an import at a given position.
+    this.imports.sort((i1, i2) => {
+      return this.comparePosition(i1.location.range.end, i2.location.range.end);
+    });
+  }
+
+  private comparePosition(p1: DocumentPosition, p2: DocumentPosition): number {
+    if (p1.line > p2.line) {
+      return 1;
+    }
+    if (p1.line < p2.line) {
+      return -1;
+    }
+    if (p1.character > p2.character) {
+      return 1;
+    }
+    if (p1.character < p2.character) {
+      return -1;
+    }
+    return 0;
+  }
+
+  private findIndexBefore(position: DocumentPosition): number {
```

Comment:
All of this can be removed if you make `ReferenceList` generic

Reply:
removed binary search code; this is obsolete